core/impl/src/main/java/com/blazebit/persistence/impl/keyset/KeysetManager.java

```diff
@@ -262,36 +262,36 @@ private void buildKeysetPredicate0(KeysetMode keysetMode, Serializable[] key, St
                     if (orderByExpr.isNullFirst() == isPrevious) {
                         // Case for previous and null first or not previous and null last
                         generateEqualPredicate = false;
-                        applyKeysetNullItem(sb, expr, false);
+                        applyKeysetNullItem(sb, expr, false, positionalOffset);
                     } else {
                         // Case for previous and null last or not previous and null first
-                        applyKeysetNullItem(sb, expr, true);
+                        applyKeysetNullItem(sb, expr, true, positionalOffset);
                     }
                 } else {
                     if (orderByExpr.isNullFirst() == isPrevious) {
                         // Case for previous and null first or not previous and null last
                         sb.append('(');
-                        applyKeysetNotNullableItem(orderByExpr, sb, i, key[i], keysetMode, !isNotLast);
+                        applyKeysetNotNullableItem(orderByExpr, sb, i, key[i], keysetMode, !isNotLast, positionalOffset);
                         sb.append(" OR ");
-                        applyKeysetNullItem(sb, expr, false);
+                        applyKeysetNullItem(sb, expr, false, positionalOffset);
                         sb.append(')');
                     } else {
                         // Case for previous and null last or not previous and null first
-                        applyKeysetNotNullableItem(orderByExpr, sb, i, key[i], keysetMode, !isNotLast);
+                        applyKeysetNotNullableItem(orderByExpr, sb, i, key[i], keysetMode, !isNotLast, positionalOffset);
                     }
                 }
             } else {
-                applyKeysetNotNullableItem(orderByExpr, sb, i, key[i], keysetMode, !isNotLast);
+                applyKeysetNotNullableItem(orderByExpr, sb, i, key[i], keysetMode, !isNotLast, positionalOffset);
             }
 
             if (isNotLast) {
                 if (generateEqualPredicate) {
                     brackets++;
                     sb.append(" OR (");
                     if (key[i] == null) {
-                        applyKeysetNullItem(sb, expr, false);
+                        applyKeysetNullItem(sb, expr, false, positionalOffset);
```

Comment:
Is this `positionalOffset` only used for parameter naming?

Reply:
This is used for positional parameters. The offset is the essentially the _maximum value_ found for a positional parameter. It is used to offset positional keyset parameters.